EIPS/eip-663.md

```diff
@@ -72,14 +89,15 @@ This has no effect on backwards compatibility because the opcodes were not previ
 Given variable `n`, which equals to `imm + 1`, for `1 <= n <= 256`:
 
     - `DUPN imm` to fail validation if `stack_height < n`.
-    - `SWAPN imm` to fail validation if `stack_height < (n + 1)`.
+    - `SWAPN imm` to fail validation if `stack_height < n + 1`.
+    - `EXCHANGE imm` to fail validation if `stack_height < n + m + 1`.
     - `DUPN imm` to increment maximum stack height of a function. Validation fails if maximum stack height exceeds limit of 1023.
-    - `DUPN imm` and `SWAPN imm` to fail at run-time if gas available is less than 3.
-    - otherwise `DUPN imm` should push the `stack[n]` item to the stack, and `SWAPN imm` should swap `stack[n + 1]` with `stack[stack.top()]`.
+    - `DUPN imm`, `SWAPN imm`, and `EXCHANGE imm` to fail at run-time if gas available is less than 3.
+    - otherwise `DUPN imm` should push the `stack[n]` item to the stack, and `SWAPN imm` should swap `stack[n + 1]` with `stack[stack.top()]` and `EXCHANGE imm` should swap `stack[n + 1]` with `stack[n + m + 1]`.
```

Comment:
These are off by one I think, because indexing of `n` is 1-based.
```suggestion
    - otherwise `DUPN imm` should push the `stack[n - 1]` item to the stack, and `SWAPN imm` should swap `stack[n]` with `stack[stack.top()]` and `EXCHANGE imm` should swap `stack[n]` with `stack[n + m]`.
```
Maybe better to just omit repeating the spec another time here.

Reply:
so just remove line 96?

Reply:
yea, why do the test cases simply repeat the spec -- we should have actual test cases